File: flut/.flutter/lib/flutter/services/text_input.dart
import 'package:flutter/services.dart';
import 'package:flut/flut/runtime.dart';
import 'package:flut/flut/object.dart';

class FlutTextInputAction extends FlutEnumObject<TextInputAction> {
  const FlutTextInputAction()
    : super('TextInputAction', const {
        'none': TextInputAction.none,
        'unspecified': TextInputAction.unspecified,
        'done': TextInputAction.done,
        'go': TextInputAction.go,
        'search': TextInputAction.search,
        'send': TextInputAction.send,
        'next': TextInputAction.next,
        'previous': TextInputAction.previous,
        'continueAction': TextInputAction.continueAction,
        'join': TextInputAction.join,
        'route': TextInputAction.route,
        'emergencyCall': TextInputAction.emergencyCall,
        'newline': TextInputAction.newline,
      });
}

class FlutTextInputType extends FlutValueObject {
  final TextInputType value;
  const FlutTextInputType(this.value) : super('TextInputType');

  @override
  Map<String, dynamic> flutEncode() {
    if (value == TextInputType.text) {
      final result = flutBaseProps();
      result['name'] = 'text';
      return result;
    }
    if (value == TextInputType.multiline) {
      final result = flutBaseProps();
      result['name'] = 'multiline';
      return result;
    }
    if (value == TextInputType.number) {
      final result = flutBaseProps();
      result['name'] = 'number';
      return result;
    }
    if (value == TextInputType.phone) {
      final result = flutBaseProps();
      result['name'] = 'phone';
      return result;
    }
    if (value == TextInputType.datetime) {
      final result = flutBaseProps();
      result['name'] = 'datetime';
      return result;
    }
    if (value == TextInputType.emailAddress) {
      final result = flutBaseProps();
      result['name'] = 'emailAddress';
      return result;
    }
    if (value == TextInputType.url) {
      final result = flutBaseProps();
      result['name'] = 'url';
      return result;
    }
    if (value == TextInputType.visiblePassword) {
      final result = flutBaseProps();
      result['name'] = 'visiblePassword';
      return result;
    }
    if (value == TextInputType.name) {
      final result = flutBaseProps();
      result['name'] = 'name';
      return result;
    }
    if (value == TextInputType.streetAddress) {
      final result = flutBaseProps();
      result['name'] = 'streetAddress';
      return result;
    }
    if (value == TextInputType.none) {
      final result = flutBaseProps();
      result['name'] = 'none';
      return result;
    }
    throw ArgumentError('Unknown TextInputType: $value');
  }

  static TextInputType? flutDecode(
    FlutRuntime runtime,
    Map<String, dynamic> data,
  ) {
    final name = runtime.unpackRequiredField<String>(data, 'name');
    if (name == 'text') return TextInputType.text;
    if (name == 'multiline') return TextInputType.multiline;
    if (name == 'number') return TextInputType.number;
    if (name == 'phone') return TextInputType.phone;
    if (name == 'datetime') return TextInputType.datetime;
    if (name == 'emailAddress') return TextInputType.emailAddress;
    if (name == 'url') return TextInputType.url;
    if (name == 'visiblePassword') return TextInputType.visiblePassword;
    if (name == 'name') return TextInputType.name;
    if (name == 'streetAddress') return TextInputType.streetAddress;
    if (name == 'none') return TextInputType.none;
    throw ArgumentError('Unknown TextInputType name: $name');
  }
}

File: flut/flutter/services/text_input.py
from typing import override

from flut._flut.engine import FlutEnumObject, FlutValueObject, _flut_pack_value
from flut._flut.runtime import _flut_unpack_required_field


class TextInputAction(FlutEnumObject):
    none: "TextInputAction"
    unspecified: "TextInputAction"
    done: "TextInputAction"
    go: "TextInputAction"
    search: "TextInputAction"
    send: "TextInputAction"
    next: "TextInputAction"
    previous: "TextInputAction"
    continueAction: "TextInputAction"
    join: "TextInputAction"
    route: "TextInputAction"
    emergencyCall: "TextInputAction"
    newline: "TextInputAction"


class _TextInputType(FlutValueObject):
    _flut_type = "TextInputType"

    def __init__(self, _name: str):
        super().__init__()
        self._name = _name

    @staticmethod
    def _flut_unpack(data: dict) -> "_TextInputType":
        return _TextInputType(
            _name=_flut_unpack_required_field(data, "name"),
        )

    @override
    def _flut_pack(self) -> dict:
        result = self._flut_base_props()
        result["name"] = self._name
        return result


class TextInputType(_TextInputType):
    text = _TextInputType("text")
    multiline = _TextInputType("multiline")
    number = _TextInputType("number")
    phone = _TextInputType("phone")
    datetime = _TextInputType("datetime")
    emailAddress = _TextInputType("emailAddress")
    url = _TextInputType("url")
    visiblePassword = _TextInputType("visiblePassword")
    name = _TextInputType("name")
    streetAddress = _TextInputType("streetAddress")
    none = _TextInputType("none")
